### ISSUE TITLE
Why average the normalization number during computing the losses?

### ISSUE THREAD

**Issue by nnop** [tianzhi0549/FCOS#228]
During computing the loss on each GPU, why did you average the normalization number (# of positive samples)? 
As I'm aware, the gradients should be computed independently on each GPU, and  the gradients are synchronized (averaged) by `torch.nn.parallel.DistributedDataParallel`.
So, averaging would make the normalization number wrong in theory. And there would have an extra synchronization.
What was your consideration?

https://github.com/tianzhi0549/FCOS/blob/dd4ca761f439a48a8c3003d1be74e9f85c050534/fcos_core/modeling/rpn/fcos/loss.py#L250-L253

**Comment by tianzhi0549** [comment]
@nnop in fact, using no averaging is wrong in theory. Averaging will make the performance consistent when you use different numbers of GPUs.

**Comment by nnop** [comment]
I didn't mean no averaging, but the averaging has been done by `torch.nn.parallel.DistributedDataParallel `.

**Comment by Yuxin-CV** [comment]
> I didn't mean no averaging, but the averaging denominator should be computed on each GPU independently.

Suppose you have 1 GPU, during 1 iter, suppose the `batch_size = 8`, the `total_loss = 8.0`, the `average_loss = 8.0 / 8 = 1` . 

Suppose now you have 4 GPU, during 1 iter, suppose the `batch_size = 8`, the `loss` in each GPU is `2.0`, then the `total_loss = 2.0 x 4 = 8.0`, the `average_loss` should also be `1`. According to your calculation, the `average_loss` is `(loss_per_GPU / batch_per_GPU) * num_GPU = (2.0 / 2) * 4 = 4`, this makes no sense.
